assumptions/html-elements.html

```diff
@@ -0,0 +1,124 @@
+<!doctype html>
+<title>HTML styles</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script><style>
+div.b {
+  all: initial;
+  direction: initial;
+  unicode-bidi: initial;
+  display: block;
+}
+
+div.c {
+  background: red;
+  background: initial;
+}
+
+span.b {
+  all: initial;
+  direction: initial;
+  unicode-bidi: initial;
+  display: inline;
+}
+</style>
+<div class="a"></div>
+<div class="b"></div>
+<div class="c"></div>
+<span class="a"></span>
+<span class="b"></span>
+<p></p>
+<ul>
+  <li>
+</ul>
+<ol>
+  <li>
+</ol>
+<table>
+  <tbody>
+    <tr>
+      <td>
+</table>
+<script>
+test(function() {
+  assert_true('all' in document.documentElement.style);
+}, "(pre-req for comparison tests) all CSS short-hand supported");
+
+test(function() {
+  assert_equals(window.getComputedStyle(document.querySelector("div.c")).backgroundColor,
+                "rgba(0, 0, 0, 0)");
+}, "(pre-req for comparison tests) initial CSS value supported");
+
+test(function() {
+  var a = document.querySelector("div.a");
+  var b = document.querySelector("div.b");
+
+  var a_styles = window.getComputedStyle(a);
+  var b_styles = window.getComputedStyle(b);
+
+  assert_equals(a_styles.length, b_styles.length, "Same properties on both #a and #b");
+
+  for (var i = 0; i < a_styles.length; i++) {
+    var property = a_styles[i];
+    assert_equals(property, b_styles[i], "Same property on #a and #b");
+    assert_equals(a_styles[property], b_styles[property], "Different value for " + property);
+  }
+}, "Compare CSS div definitions (only valid if pre-reqs pass)");
+
+test(function() {
+  var a = document.querySelector("span.a");
+  var b = document.querySelector("span.b");
+
+  var a_styles = window.getComputedStyle(a);
+  var b_styles = window.getComputedStyle(b);
+
+  assert_equals(a_styles.length, b_styles.length, "Same properties on both #a and #b");
+
+  for (var i = 0; i < a_styles.length; i++) {
+    var property = a_styles[i];
+    assert_equals(property, b_styles[i], "Same property on #a and #b");
```

Comment:
I'm not quite sure if all of this is guaranteed, but I'm not going to worry about it until someone complains.